apps/heft/src/plugins/TimelineTracePlugin.ts

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { FileWriter } from '@rushstack/node-core-library';
+
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { IHeftPlugin } from '../pluginFramework/IHeftPlugin';
+import { HeftSession } from '../pluginFramework/HeftSession';
+import { IHeftLifecycle } from '../pluginFramework/HeftLifecycle';
+import { HookInterceptor, SyncHook, Tap, TapOptions } from 'tapable';
+import { IBuildStageContext } from '../stages/BuildStage';
+import { ITestStageContext } from '../stages/TestStage';
+import { ICleanStageContext } from '../stages/CleanStage';
+
+const PLUGIN_NAME: 'TimelineTracePlugin' = 'TimelineTracePlugin';
+
+/**
+ * This plugin generates a timeline trace of the Heft execution in the Chrome Trace Events format.
+ */
+export class TimelineTracePlugin implements IHeftPlugin {
+  public readonly pluginName: 'TimelineTracePlugin' = PLUGIN_NAME;
+
+  public apply(heftSession: HeftSession, heftConfiguration: HeftConfiguration): void {
+    const traceFileName: string = `${heftConfiguration.buildFolder}/heft.trace.${Date.now()}.json`;
```

Comment:
Same as my [comment](https://github.com/microsoft/rushstack/pull/2569/files#r605366585) on https://github.com/microsoft/rushstack/pull/2569. Seems like this should come from the user.